tests/testSpherePoint.py

```diff
@@ -0,0 +1,751 @@
+        """
+        # The problem was that the coordinate is less than epsilon
+        # close to RA == 0 and bounds checking was getting a
+        # negative RA.
+        point = SpherePoint(afwGeom.Point3D(0.6070619982, -1.264309928e-16, 0.7946544723))
+
+        self.assertEqual(point[0].asDegrees(), 0.0)
+
+    def testGetLatitudeValue(self):
+        """Test if getLatitude() returns the expected value.
+        """
+        for lon, lat in self._dataset:
+            point = SpherePoint(lon, lat)
+            self.assertIsInstance(point.getLatitude(), afwGeom.Angle)
+            if point.isFinite():
+                self.assertGreaterEqual(point.getLatitude().asDegrees(), -90.0)
+                self.assertLessEqual(point.getLatitude().asDegrees(), 90.0)
+                self.assertAnglesNearlyEqual(lat, point.getLatitude())
+            # Behavior for non-finite points is undefined; depends on internal data representation
+
+    def testGetVectorValue(self):
+        """Test if getVector() returns the expected value.
+
+        The test includes conformance to vector-angle conventions.
+        """
+        pointList = [
+            ((0.0, 0.0), afwGeom.Point3D(1.0, 0.0, 0.0)),
+            ((90.0, 0.0), afwGeom.Point3D(0.0, 1.0, 0.0)),
+            ((0.0, 90.0), afwGeom.Point3D(0.0, 0.0, 1.0)),
+        ]
+
+        for lonLat, vector in pointList:
+            # Convert to Point3D.
+            point = SpherePoint(lonLat[0]*degrees, lonLat[1]*degrees)
+            newVector = point.getVector()
+            self.assertIsInstance(newVector, afwGeom.Point3D)
+            for oldElement, newElement in zip(vector, newVector):
+                self.assertAlmostEqual(oldElement, newElement)
+
+            # Convert back to spherical.
+            newLon, newLat = SpherePoint(newVector)
+            self.assertAlmostEqual(newLon.asDegrees(), lonLat[0])
+            self.assertAlmostEqual(newLat.asDegrees(), lonLat[1])
+
+        # Try some un-normalized ones, too.
+        pointList = [
+            ((0.0, 0.0), afwGeom.Point3D(1.3, 0.0, 0.0)),
+            ((90.0, 0.0), afwGeom.Point3D(0.0, 1.2, 0.0)),
+            ((0.0, 90.0), afwGeom.Point3D(0.0, 0.0, 2.3)),
+            ((0.0, 0.0), afwGeom.Point3D(0.5, 0.0, 0.0)),
+            ((90.0, 0.0), afwGeom.Point3D(0.0, 0.7, 0.0)),
+            ((0.0, 90.0), afwGeom.Point3D(0.0, 0.0, 0.9)),
+        ]
+
+        for lonLat, vector in pointList:
+            # Only convert from vector to spherical.
+            point = SpherePoint(vector)
+            newLon, newLat = point
+            self.assertAlmostEqual(lonLat[0], newLon.asDegrees())
+            self.assertAlmostEqual(lonLat[1], newLat.asDegrees())
+            self.assertAlmostEqual(1.0, point.getVector().distanceSquared(afwGeom.Point3D(0.0, 0.0, 0.0)))
+
+        # Ill-defined points should be all NaN after normalization
+        pointList = [
+            afwGeom.Point3D(0.5, nan, 0.0),
+            afwGeom.Point3D(0.5, 0.0, -inf),
```

Comment:
I suggest a full test of nan and +/- inf appearing once in each of x, y and z. I realize x and y are probably handled fairly symmetrically, but z is handled differently and might as well cover all bases. Personally I'd use a loop to replace each element in turn by (nan, inf or -inf) but I realize loops are not necessarily well liked by all test authors.